test/test_torch.py

```diff
@@ -7290,6 +7290,55 @@ def run_test(p):
         with self.assertRaisesRegex(RuntimeError, "chain_matmul: Expected one or more matrices"):
             torch.chain_matmul()
 
+    # no CUDA LU yet, used for torch.det())
+    @onlyCPU
+    @skipCPUIfNoLapack
+    @dtypes(torch.complex64, torch.complex128)
+    def test_det_complex_only(self, device, dtype):
+        scalar_dtype = {
+            torch.complex64: torch.float32,
+            torch.complex128: torch.float64
+        }
+
+        def run_test(*sizes):
+            x = torch.rand(*sizes, device=device, dtype=dtype)
+            x = x.view(-1, sizes[-2], sizes[-1])
```

Comment:
why are you sending 4d sizes if you immediately turn them into 3d?

Reply:
I only create a view for easier comparison against NumPy per each matrix in the batch.